### PR TITLE
Build and add semantic-mocha mocha wrapper

### PR DESCRIPTION
**Pull request by rachidlamouri**
## Motivation

Instead of documenting test practices, why not enforce them in code? [Why not Zoidberg?](https://preview.redd.it/z1wzychvown21.jpg?width=620&format=pjpg&auto=webp&s=1079809fe0136e07712db54b1910d253845e1369)

## Description

Created a wrapper around mocha to provide a better structure for tests.

Reviewer recommendation:
- Skip the first two commits. The semantic-mocha library is dense and is currently all in one file 😁.
- The updates to databaseUtil and gameStateUtil will provide some better context
- For more information:
  - Look at the README.md in the `semantic-mocha/` folder
  - Look at the abstract examples in the semantic-mocha tests `semantic-mocha/test/examples/`

### CODE REVIEW

**Comment by rachidlamouri** [comment]
The merge commit dropped the `databaseUtil.test.ts` changes, so I modified the merge commit to reapply the changes.

File: package.json [hunk 1]
```diff
@@ -9,6 +9,7 @@
     "lint:ts": "eslint --ext .ts,.js .",
     "lint:yaml": "yamllint --ignore=node_modules/**/*.yaml **/*.yaml",
     "test:e2e": "mocha \"tests/e2e/**/*.test.ts\"",
+    "test:semantic-mocha": "nyc --extension .ts mocha --config semantic-mocha/tests/.mocharc.js \"semantic-mocha/tests/**/*.test.ts\"",
```
Comment (jshemas) [inline package.json:12]: Should `semantic-mocha` have it's own `package.json` with this command?

File: semantic-mocha/README.md [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+
+Wraps mocha in an opinionated framework to provide a rigid structure for tests.
+The library is a work in progress and will need to be modified as new use cases arise.
+
+- [Roadmap](#roadmap)
+- [Current API](#current-api)
+
+## Roadmap
+
+- Support mocha's .only and .skipped
```
Comment (jshemas) [inline semantic-mocha/README.md:11]: Should this line be removed?

File: package.json (hunk 1 above)
Comment (rachidlamouri) [reply package.json:12]: Nah

File: semantic-mocha/README.md (hunk 2 above)
Comment (rachidlamouri) [reply semantic-mocha/README.md]: Updated

**Review by jshemas — APPROVED** [review]

**Pull request merged**